hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -1319,6 +1321,8 @@ private void initializeClientOnCluster(UUID targetClusterId) {
     }
 
     private void connectTpcPorts(TcpClientConnection connection, List<Integer> tpcPorts, byte[] tpcToken) {
+        List<Integer> targetTpcPorts = getTargetTpcPorts(tpcPorts, client.getClientConfig().getTpcConfig());
+
         TpcChannelConnector connector = new TpcChannelConnector(
```

Comment:
I guess it would be good to have a integration test (I mean when we spin a client, not a class based unit test) to test we connect to selected ports only.